lib/membrane/video_compositor/handler/callback_ctx.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Membrane.VideoCompositor.Handler.CallbackContext do
+  @moduledoc """
+  Structure representing a common part of the context
+  for all of the callbacks.
+  """
+  alias Membrane.{Pad, Time}
+  alias Membrane.VideoCompositor.Scene
+
+  defmacro __using__(custom_fields) do
+    default_fields =
+      quote do
+        [
+          input_pads: list(Pad.ref_t()),
+          scenes_queue: list({scene :: Scene.t(), start_scene_timestamp :: Time.t()}),
```

Comment:
[NIT] Usually in priority queues, the priority level is the first argument. `scenes_queue` is just the priority queue with the lowest `start_scene_timestamp` and `scenes` having the highest priority. 
```suggestion
          scenes_queue: list({start_scene_timestamp :: Time.t(), scene :: Scene.t()}),
```